### PR TITLE
feat: create dropdown option for custom api token creation

### LINKED ISSUE #1924: Tokens: Add Custom Token Dropdown
If the feature flag is ON, then a new "Generate Custom Token" option will appear under the Generate New Token CTA.

Else the existing "read/write token" for Generate New Token CTA will continue to appear.

Clicking on the `Generate Custom Token` dropdown option, opens an overlay.

### PR DESCRIPTION
**Pull request by Palakp41**
Closes #1924 
Added custom api token dropdown option behind the feature flag and for CLOUD users only.

Screen Shot 2021-07-20 at 12 29 25 PM

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 52 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2085)

File: src/authorizations/components/CustomApiTokenOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <Overlay.Container maxWidth={500}>
+      <Overlay.Header
+        title="Generate a Personal Api Token"
+        onDismiss={handleDismiss}
+      />
+    </Overlay.Container>
+  )
+}
+
+export default CustomApiTokenOverlay
```
Comment (hoorayimhelping) [inline src/authorizations/components/CustomApiTokenOverlay.tsx:24]: can you make this a named export please?

named exports automatically get tree-shaken by webpack which reduces bundle size. It also removes ambiguities when importing

File: src/authorizations/components/GenerateTokenDropdown.test.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+})
+
+import GenerateTokenDropdown from 'src/authorizations/components/GenerateTokenDropdown'
+
+const setup = () => {
+  return renderWithReduxAndRouter(<GenerateTokenDropdown />)
+}
+
+describe('GenerateTokenDropdown', () => {
+  it('renders the dropdown', () => {
```
Comment (hoorayimhelping) [inline src/authorizations/components/GenerateTokenDropdown.test.tsx:28]: follow up: maybe write a test for when `CLOUD` is false so we can be confident this won't accidentally get turned on in OSS

File: src/authorizations/components/GenerateTokenDropdown.tsx [hunk 3]
```diff
@@ -21,38 +25,59 @@ const GenerateTokenDropdown: FC<GenerateTokenProps> = ({history}) => {
...
+            id={customApiOption}
+            key={customApiOption}
+            value={customApiOption}
+            onClick={handleSelect}
+          >
+            {customApiOption}
+          </Dropdown.Item>
+        </>
+      )
+    } else {
```
Comment (hoorayimhelping) [inline src/authorizations/components/GenerateTokenDropdown.tsx:78]: no need for this `else` since the `if` statement returns

**Review by hoorayimhelping — APPROVED** [review]
looks good, just a couple of small nitpicks

File: src/authorizations/components/GenerateTokenDropdown.test.tsx (hunk 2 above)
Comment (Palakp41) [reply src/authorizations/components/GenerateTokenDropdown.test.tsx:28]: Will add this use case in next pr as it is a bit more complicated to mock the `CLOUD` constant to false.

File: src/authorizations/components/GenerateTokenDropdown.test.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  })
+  it('renders the dropdown options', () => {
+    const {queryByText} = setup()
+
+    fireEvent.click(screen.getByTestId('dropdown-button--gen-token'))
+
+    expect(queryByText('Read/Write Token')).toBeVisible()
+    expect(queryByText('All Access Token')).toBeVisible()
+  })
+})
```
Comment (hoorayimhelping) [inline src/authorizations/components/GenerateTokenDropdown.test.tsx:28]: i think we can get rid of this file, right?

**Review by hoorayimhelping — APPROVED** [review]
:+1:

File: src/authorizations/components/GenerateTokenDropdown.test.tsx (hunk 4 above)
Comment (Palakp41) [reply src/authorizations/components/GenerateTokenDropdown.test.tsx:28]: We can keep it, it tests the current functionality of the dropdown. Wdyt?

**Pull request merged**